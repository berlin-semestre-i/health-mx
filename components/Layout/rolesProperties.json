{
  "medic": {
    "placeholder": "Buscar derechohabiente o tarea",
    "background": "#3d5170"
  },
  "beneficiary": {
    "placeholder": "Buscar tarea o servicio",
    "background": "#0057A3"
  },
  "nurse": {
    "placeholder": "Buscar derechohabiente",
    "background": "#01493D"
  },
  "admin": {
    "placeholder": "Buscar usuario",
    "background": "#561A9B"
  },
  "error": {
    "placeholder": "Error: you have no role",
    "background": "#F65D61"
  }
}