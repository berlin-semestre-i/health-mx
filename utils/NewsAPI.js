import NewsAPI from 'newsapi'
import getConfig from 'next/config'

const config = getConfig()
const {
  NEWS_API_KEY,
} = (config) ? config.publicRuntimeConfig : {}
const newsapi = new NewsAPI(NEWS_API_KEY || 'none')

export default {
  getData() {
    return newsapi.v2.everything({
      q: 'salud',
      language: 'es',
      pageSize: 4,
    })
  },
}