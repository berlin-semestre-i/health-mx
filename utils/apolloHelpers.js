import get from 'lodash/get'

/**
 * Returns the URL of GraphQL proxy based on actual domain
 *
 * @param {Object||undefined} req: Request object
 * @return {String} The URL of the GraphQL proxy
 */
const getGraphQLProxyURL = (req) => {
  if (global.GraphQLProxy) {
    return global.GraphQLProxy
  }

  const domain = process.browser ?
    window.location.href.split('/')[2] :
    get(req, 'headers["x-forwarded-host"]', get(req, 'headers.host', ''))

  const protocol = process.browser ?
    window.location.href.split(':')[0] :
    get(req, 'protocol', 'https')

  global.GraphQLProxy = `${protocol}://${domain}/graphql`

  return global.GraphQLProxy
}

export { getGraphQLProxyURL }
