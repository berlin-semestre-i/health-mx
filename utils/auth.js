import getConfig from 'next/config'
import redirect from './redirect'
import { getCookie, removeCookie } from './session'
import { Auth } from 'aws-amplify'

const config = getConfig()
const {
  AWS_REGION,
  USER_POOL_ID,
  USER_POOL_CLIENT_ID,
} = (config) ? config.publicRuntimeConfig : {}

const getCookieSessionName = () => {
  const cookieName = 'cognito-access-token'

  return cookieName
}

export const amplifyConfig = {
  region: AWS_REGION,
  userPoolId: USER_POOL_ID,
  userPoolWebClientId: USER_POOL_CLIENT_ID,
  authenticationFlowType: 'USER_SRP_AUTH',
}

export const signOut = (ctx = {}) => {
  if (process.browser) {
    removeCookie(getCookieSessionName())
    redirect('/', ctx)
  }
}

export const getAuthToken = () => getCookie(getCookieSessionName())

export const isAuthenticated = async ctx => {
  let hasAuthedUser
  try {
    hasAuthedUser = !!(await Auth.currentAuthenticatedUser())
  } catch (err) {
    hasAuthedUser = false
  }
  if (!!getAuthToken(ctx) && hasAuthedUser) return true
  else return false
}

export const redirectIfAuthenticated = async ctx => {
  const isLoggedIn = await isAuthenticated(ctx)
  if (process.browser && isLoggedIn) {
    redirect('/medic', ctx)
    return true
  }
  return false
}

export const redirectIfNotAuthenticated = async ctx => {
  const isLoggedIn = await isAuthenticated(ctx)
  if (process.browser && !isLoggedIn) {
    redirect('/', ctx)
    return true
  }
  return false
}