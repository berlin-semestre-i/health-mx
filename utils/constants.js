const testsConstants = {
  userRole: 'medic',
  userGender: 'male',
  newMedication: {
    dose: '20ml',
    medicine: 'Dramamine',
    frequency: '8 hrs',
    duration: '4 días',
  },
  newStudy: {
    study: 'Rayos X',
    indications: 'Sin indicaciones',
  },
  errorMessage: 'Llena todos los campos.',
  somatometry: {
    nurse: 'Rodrigo Fernández',
    height: 169,
    weight: 63,
    pulse: 68,
    imc: 22.06,
    temperature: 36,
  },
}

const defaultManAvatar = '/static/images/man-avatar.png'
const defaultWomanAvatar = '/static/images/woman-avatar.png'

export { testsConstants, defaultManAvatar, defaultWomanAvatar }