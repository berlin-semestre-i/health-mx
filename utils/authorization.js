import { medic, beneficiary, nurse, error, admin } from '../components/Layout/rolesProperties.json'

const getPropertiesFromRole = (userRole) => {
  if ( userRole === 'medic' ) {
    return medic
  }
  else if ( userRole === 'beneficiary' ) {
    return beneficiary
  }
  else if ( userRole === 'nurse' ) {
    return nurse
  }
  else if ( userRole === 'admin' ) {
    return admin
  }
  else {
    return error
  }
}

export { getPropertiesFromRole }