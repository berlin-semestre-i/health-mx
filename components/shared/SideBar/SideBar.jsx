import React from 'react'
import Link from 'next/link'
import { Menu, Image, Icon } from 'semantic-ui-react'
import styled from 'styled-components'
import { getPropertiesFromRole } from '../../../utils/auth'

class SideBar extends React.PureComponent {
  state = {
    activeItem: 'home',
  }

  handleItemClick = (e, { name }) => this.setState({ activeItem: name })

  render() {
    const { userRole } = this.props
    const { activeItem } = this.state
    const roleProperties = getPropertiesFromRole(userRole)

    return (
      <SideMenu
        name="sideMenu"
        vertical
        fixed="left"
        background={roleProperties.background}
        icon
      >
        <Link as={`/${roleProperties.name}`} href={`/${userRole}`}>
          <HealthImage src="../../../static/images/Health-Bar-Logo.png"/>
        </Link>
        <Menu.Item
          name="home"
          active={activeItem === 'home'}
          onClick={this.handleItemClick}
        >
          <Icon name="home" size="large"/>
        </Menu.Item>
        <Menu.Item
          name="user"
          active={activeItem === 'user'}
          onClick={this.handleItemClick}
        >
          <Icon name="user" size="large"/>
        </Menu.Item>
      </SideMenu>
    )
  }
}

export default SideBar

const HealthImage = styled(Image)`
  && {
    padding: 15px 21px 36px 21px;
    cursor: pointer;
  }
`

const SideMenu = styled(Menu)`
  &&&& {
    border: none;
    -webkit-box-shadow: 0 0 10px 1px #e7e9ed;
    box-shadow: 0 0 10px 1px #e7e9ed;
    background-color: ${props => props.background};
  }
`
