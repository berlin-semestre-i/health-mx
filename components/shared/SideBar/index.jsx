import React from 'react'
import Link from 'next/link'
import { withRouter } from 'next/router'
import { Menu, Image, Icon, Popup } from 'semantic-ui-react'
import styled from 'styled-components'
import { getPropertiesFromRole } from '../../../utils/auth'
import { defaultManAvatar, defaultWomanAvatar } from '../../../utils/constants'

class SideBar extends React.PureComponent {
  render() {
    const { userRole, router, userAvatar, userGender } = this.props
    const roleProperties = getPropertiesFromRole(userRole)
    const genderDefaultAvatar = userGender === 'male' ? defaultManAvatar : defaultWomanAvatar
    const route = router.route

    return (
      <SideMenu
        name="sideMenu"
        vertical
        fixed="left"
        background={roleProperties.background}
        icon
      >
        <Link href={`/${userRole}`}>
          <HealthImage src="../../../static/images/Health-Bar-Logo.png"/>
        </Link>
        <Link href={`/${userRole}`}>
          <Menu.Item
            name="home"
            active={`/${userRole}` === route}
          >
            <Popup
              flowing
              inverted
              size="tiny"
              position="top right"
              horizontalOffset={12}
              trigger = {
                <Icon name="home" size="large"/>
              }
              content="Inicio"
            />
          </Menu.Item>
        </Link>
        { roleProperties.menuItems.map(item => (
          <Link href={`/${userRole}${item.address}`} key={item.id}>
            <Menu.Item
              name={item.name}
              active={`/${userRole}${item.address}` === route}
            >
              <Popup
                flowing
                inverted
                size="tiny"
                position="top right"
                horizontalOffset={12}
                trigger = {
                  <Icon name={item.icon} size="large"/>
                }
                content={item.description}
              />
            </Menu.Item>
          </Link>
        ))}
        <Notifications name="notifications">
          <Popup
            flowing
            inverted
            size="tiny"
            position="top right"
            horizontalOffset={12}
            trigger = {
              <Icon name="bell" size="large"/>
            }
            content="Notificaciones"
          />
        </Notifications>
        <Tips name="tips">
          <Popup
            flowing
            inverted
            size="tiny"
            position="top right"
            horizontalOffset={12}
            trigger = {
              <Icon name="question circle" size="large"/>
            }
            content="Ayuda"
          />
        </Tips>
        <Avatar
          name="userAvatar"
          src={userAvatar ? userAvatar : genderDefaultAvatar}
          circular
        />
      </SideMenu>
    )
  }
}

export default withRouter(SideBar)

const HealthImage = styled(Image)`
  && {
    margin: 15px 21px 36px 21px;
    width: 42px;
    height: 40px;
    cursor: pointer;
  }
`

const SideMenu = styled(Menu)`
  &&&& {
    display: flex;
    border: none;
    -webkit-box-shadow: 0 0 10px 1px #e7e9ed;
    box-shadow: 0 0 10px 1px #e7e9ed;
    background-color: ${props => props.background};
  }
`

const Avatar = styled(Image)`
  &&&& {
    background-color: #FFFFFF;
    width: 48.8px;
    height: 48.8px;
    margin: 35.6px 18px 50.6px 21px;
    align-self: flex-end;
    border: solid 2px #FFFFFF;
  }
`

const Notifications = styled(Menu.Item)`
  &&&& {
    margin-top: auto;
    align-self: flex-end;
  }
`

const Tips = styled(Menu.Item)`
  &&&& {
    margin-bottom: 0;
    align-self: flex-end;
  }
`