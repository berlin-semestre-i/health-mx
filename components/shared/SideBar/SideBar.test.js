import React from 'react'
import { shallow, mount } from 'enzyme'
import SideBar from './index'
import { testsConstants, defaultManAvatar, defaultWomanAvatar } from '../../../utils/constants'
import { medic, beneficiary, nurse, admin, error } from '../../Layout/rolesProperties.json'
import 'jest-styled-components'

describe('SideBar', () => {
  it('renders', () => {
    const wrapper = shallow(
      <SideBar userRole={testsConstants.userRole} userGender={testsConstants.userGender}/>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should have medic properties when medic role', () => {
    const wrapper = mount(
      <SideBar userRole="medic" userGender={testsConstants.userGender} router={jest.fn()}/>
    )
    expect(wrapper).toHaveStyleRule(
      'background-color',
      medic.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })
  it('should have beneficiary properties when beneficiary role', () => {
    const wrapper = mount(
      <SideBar userRole="beneficiary" userGender={testsConstants.userGender} router={jest.fn()}/>
    )
    expect(wrapper).toHaveStyleRule(
      'background-color',
      beneficiary.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })

  it('should have nurse properties when nurse role', () => {
    const wrapper = mount(
      <SideBar userRole="nurse" userGender={testsConstants.userGender} router={jest.fn()}/>
    )
    expect(wrapper).toHaveStyleRule(
      'background-color',
      nurse.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })

  it('should have admin properties when admin role', () => {
    const wrapper = mount(
      <SideBar userRole="admin" userGender={testsConstants.userGender} router={jest.fn()}/>
    )
    expect(wrapper).toHaveStyleRule(
      'background-color',
      admin.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })

  it('should have error properties when error', () => {
    const wrapper = mount(
      <SideBar userRole="error" userGender={testsConstants.userGender} router={jest.fn()}/>
    )
    expect(wrapper).toHaveStyleRule(
      'background-color',
      error.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })

  it('should render userAvatar if provided', () => {
    const wrapper = mount(
      <SideBar userAvatar={defaultWomanAvatar} userRole="error" router={jest.fn()}/>
    )
    expect(wrapper.find('img[name="userAvatar"]').prop('src')).toEqual(defaultWomanAvatar)
  })

  it('should render default woman avatar when no userAvatar found and user is female', () => {
    const wrapper = mount(
      <SideBar userRole="error" userGender="female" router={jest.fn()}/>
    )
    expect(wrapper.find('img[name="userAvatar"]').prop('src')).toEqual(defaultWomanAvatar)
  })

  it('should render default man avatar when no userAvatar found and user male', () => {
    const wrapper = mount(
      <SideBar userRole="error" userGender="male" router={jest.fn()}/>
    )
    expect(wrapper.find('img[name="userAvatar"]').prop('src')).toEqual(defaultManAvatar)
  })
})