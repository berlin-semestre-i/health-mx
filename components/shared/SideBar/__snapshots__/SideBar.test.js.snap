// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders 1`] = `
<SideBar__SideMenu
  background="#F65D61"
  fixed="left"
  icon={true}
  name="sideMenu"
  vertical={true}
>
  <SideBar__HealthImage
    src="../../../static/images/Health-Bar-Logo.png"
  />
  <Link
    as="/admin/home"
    href="/admin"
  >
    <MenuItem
      active={true}
      name="home"
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="home"
        size="large"
      />
    </MenuItem>
  </Link>
  <Link
    as="/doctor/hello"
    href="/doctor/hola"
  >
    <MenuItem
      active={false}
      name="user"
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="user"
        size="large"
      />
    </MenuItem>
  </Link>
</SideBar__SideMenu>
`;
